packages/@lwc/engine/src/framework/renderer.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+export type HostNode = any;
+export type HostElement = any;
+
+export interface Renderer<N = HostNode, E = HostElement> {
+    syntheticShadow: boolean;
+    insert(node: N, parent: E, anchor: N | null): void;
+    remove(node: N, parent: E): void;
+    createElement(tagName: string, namespace?: string): E;
+    createText(content: string): N;
+    nextSibling(node: N): N | null;
+    attachShadow(
+        element: E,
+        options: { mode: 'open' | 'closed'; delegatesFocus?: boolean; [key: string]: any }
```

Comment:
Try to use the standard DOM types as much as possible.
```suggestion
        options: ShadowRootInit
```